Source/creature_internal.cpp

```diff
@@ -79,13 +80,19 @@ TOperator* creature_internal::_get_genetic_operator(TMetadata** operator_metadat
 }
 
 crossover* creature_internal::_get_crossover(crossover_metadata** metadata, size_t num_crossovers) {
-    return _get_genetic_operator<crossover, crossover_metadata>(metadata, num_crossovers,
-            create_genetic_operator_a<crossover, crossover_metadata>(genetic_factory::create_crossover));
+    genetic_factory& factory = genetic_factory::get_instance();
+    create_genetic_operator_a<crossover, crossover_metadata>  create_crossover = [&factory](crossover_metadata* metadatai) {
```

Comment:
what does `[&factory]` mean, specifically the `[ ]` part?

Reply:
[ ] before a lambda expression is a 'capture clause'. Since lambda expressions define a new scope C++ requires you to declare what members of the containing scope you would like to 'capture' or have available in the lambda scope. In this case I am capturing a reference to the genetic_factory variable (&factory).